crates/runtime/src/memory.rs

```diff
@@ -359,7 +366,12 @@ impl Memory {
                 make_accessible,
                 ..
             } => {
-                if new_size > maximum.unwrap_or(WASM_MAX_PAGES) {
+                if let Some(maximum) = maximum {
+                    if new_size > maximum {
+                        return None;
+                    }
+                }
+                if new_size > WASM_MAX_PAGES - 1 {
```

Comment:
Like above, could the existing check be left as-is and this added as a new check with a `FIXME` which we can easily delete in the future?